embassy-stm32/src/hrtim/traits.rs

```diff
@@ -0,0 +1,200 @@
+use crate::rcc::sealed::RccPeripheral;
+use crate::time::Hertz;
+
+#[derive(Clone, Copy)]
+pub(crate) enum HighResolutionControlPrescaler {
+    Div1,
+    Div2,
+    Div4,
+    Div8,
+    Div16,
+    Div32,
+    Div64,
+    Div128,
+}
+
+impl From<HighResolutionControlPrescaler> for u32 {
+    fn from(val: HighResolutionControlPrescaler) -> Self {
+        match val {
+            HighResolutionControlPrescaler::Div1 => 1,
+            HighResolutionControlPrescaler::Div2 => 2,
+            HighResolutionControlPrescaler::Div4 => 4,
+            HighResolutionControlPrescaler::Div8 => 8,
+            HighResolutionControlPrescaler::Div16 => 16,
+            HighResolutionControlPrescaler::Div32 => 32,
+            HighResolutionControlPrescaler::Div64 => 64,
+            HighResolutionControlPrescaler::Div128 => 128,
+        }
+    }
+}
+
+impl From<HighResolutionControlPrescaler> for u8 {
+    fn from(val: HighResolutionControlPrescaler) -> Self {
```

Comment:
it is very strange that converting to u32 gives the prescaler value and converting to u8 gives the register value. Perhaps this is a missing enum that could be added to stm32-metapac?

Reply:
I will add it later, but I would not like to wait for another metapac release.

Reply:
Also, this code doesn't leave this crate.

Reply:
:smiling_face_with_tear: